mysql/datadog_checks/mysql/statements.py

```diff
@@ -0,0 +1,126 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from contextlib import closing
+
+import pymysql
+
+from datadog_checks.base import AgentCheck, is_affirmative
+from datadog_checks.base.utils.db.sql import compute_sql_signature
+from datadog_checks.base.utils.db.statement_metrics import StatementMetrics
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+
+METRICS = {
+    'count': ('mysql.queries.count', AgentCheck.count),
+    'errors': ('mysql.queries.errors', AgentCheck.count),
+    'time': ('mysql.queries.time', AgentCheck.count),
+    'select_scan': ('mysql.queries.select_scan', AgentCheck.count),
+    'select_full_join': ('mysql.queries.select_full_join', AgentCheck.count),
+    'no_index_used': ('mysql.queries.no_index_used', AgentCheck.count),
+    'no_good_index_used': ('mysql.queries.no_good_index_used', AgentCheck.count),
+    'lock_time': ('mysql.queries.lock_time', AgentCheck.count),
+    'rows_affected': ('mysql.queries.rows_affected', AgentCheck.count),
+    'rows_sent': ('mysql.queries.rows_sent', AgentCheck.count),
+    'rows_examined': ('mysql.queries.rows_examined', AgentCheck.count),
+}
+
+
+DEFAULT_METRIC_LIMITS = {
+    # TODO: Reduce limits after usage patterns are better understood
+    'count': (10000, 10000),
+    'errors': (10000, 10000),
+    'time': (10000, 10000),
+    'select_scan': (200, 200),
+    'select_full_join': (200, 200),
+    'no_index_used': (200, 200),
+    'no_good_index_used': (200, 200),
+    'lock_time': (200, 200),
+    'rows_affected': (500, 500),
+    'rows_sent': (500, 500),
+    'rows_examined': (500, 500),
+}
+
+class MySQLStatementMetrics(StatementMetrics):
```

Comment:
Should we use `StatementMetrics` as a library instead of through inheritance? I'm not sure inheritance makes sense in this case. This code isn't interacting with the instance config or the database client.

Reply:
Yeah I was thinking the same today actually. State is required so I think it still makes sense as a class, but I will not use inheritance.